link-check.js

```diff
@@ -23,37 +33,48 @@ function allMatches(str, regex) {
 
 let invalid = 0;
 
-for (const file of files) {
-  const filePretty = chalk.bold(
-    path
-      .relative(__dirname, file)
-      .replace(/^public/, "")
-      .replace(/index.html$/, "")
-  );
-  const contents = fs.readFileSync(file, "utf8");
-  if (contents.indexOf("Postgraphile") >= 0) {
-    invalid++;
-    console.error(
-      `${filePretty} mentions 'Postgraphile'; please change to 'PostGraphile' or 'postgraphile' for consistency.`
+const fileLinks = files
+  .map(file => {
+    const filePretty = chalk.bold(
+      path
+        .relative(__dirname, file)
+        .replace(/^public/, "")
+        .replace(/index.html$/, "")
     );
-  }
-  const links = allMatches(contents, /<a[^>]+href="([^"]+)"/g);
-  for (const link of links) {
+    const contents = fs.readFileSync(file, "utf8");
+    if (contents.indexOf("Postgraphile") >= 0) {
+      invalid++;
+      console.error(
+        `${filePretty} mentions 'Postgraphile'; please change to 'PostGraphile' or 'postgraphile' for consistency.`
+      );
+    }
+
+    // WARNING! Cardinal sin below, shield eyes
+    return allMatches(contents, /<a[^>]+href="([^"]+)"/g).map(link => {
+      return {
+        link: entities.decode(link),
+        filePretty,
+      };
+    });
+  })
+  .reduce((flattened, toFlatten) => flattened.concat(toFlatten), []);
+
+pMap(
+  fileLinks,
+  async ({ filePretty, link }) => {
     const trimmed = link.replace(/[?#].*$/, "");
-    if (trimmed.match(/\.(js|css|png|svg|webmanifest)$/)) {
+
+    if (trimmed.match(/\.(css|png|svg|webmanifest)$/)) {
       // Meh, resources
-      continue;
+      return;
     }
-    const isGitHub = trimmed.match(/^https?:\/\/(?:www\.)?github\.com/);
-    const isLearnDotGraphile = trimmed.match(
-      /^https?:\/\/learn\.graphile\.org\//
-    );
-    const isGraphile = trimmed.match(/^https?:\/\/graphile\.(org|meh)/);
-    const isLocalhost = trimmed.match(
-      /^https?:\/\/(localhost|127\.0\.0\.1|0\.0\.0\.0):(?:[^5]|5[^0]|50[^0]|500[^0])/
+    const isAbsolute = /^[a-z]+:\/\//.test(trimmed);
+    const isGraphile = /^https?:\/\/graphile\.(org|meh)/.test(trimmed);
```

Comment:
replaced most `String#match` calls with `RegExp.test`, they are faster and easier to read (imo).